codebuild/canary/CanaryWrapper_MetricFunctions.py

```diff
@@ -0,0 +1,52 @@
+# Contains all of the metric reporting functions for the Canary Wrappers
+
+# Needs to be installed prior to running
+import psutil
+
+
+cache_cpu_psutil_process = None
+
+
+def get_metric_total_cpu_usage(psutil_process: psutil.Process):
+    global cache_cpu_psutil_process
+
+    try:
+        if (psutil_process == None):
+            print("ERROR - No psutil.process passed! Cannot gather metric!", flush=True)
+            return None
+        # We always need to skip the first CPU poll
+        if (cache_cpu_psutil_process != psutil_process):
+            psutil.cpu_percent(interval=None)
+            cache_cpu_psutil_process = psutil_process
+            return None
+        return psutil.cpu_percent(interval=None)
+    except Exception as e:
+        print("ERROR - exception occurred gathering metrics!")
```

Comment:
There are three separate instances of this pattern of exception handling. Let's call out which part of the metrics gathering the metrics exception was encountered in. e.g. "exception occurred gathering memory usage value metrics"